src/Build/BackEnd/Components/ProjectCache/ProjectCachePluginBase.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.Build.Experimental.ProjectCache
 {
     /// <summary>
     ///     Only one plugin instance can exist for a given BuildManager BeginBuild / EndBuild session.
+    ///     Any exceptions thrown by the plugin will cause MSBuild to fail the build.
```

Comment:
@Forgind I added an extra comment to make it clear what MSBuild does on plugin exceptions. I think it's benign enough to have it in this PR, but let me know if you think otherwise.